File: project/recruiters/backend/services/positions-manager-service/src/main/java/com/jaruiz/casarrubios/recruiters/services/posmanager/business/exceptions/PositionNotFoundException.java
package com.jaruiz.casarrubios.recruiters.services.posmanager.business.exceptions;

import java.io.Serial;

import org.jboss.logging.Logger;

public class PositionNotFoundException extends RuntimeException {
    private static final Logger logger = Logger.getLogger(PositionNotFoundException.class);

    @Serial private static final long serialVersionUID = 1L;
    private final String code;

    public PositionNotFoundException(long positionId) {
        super("Position not found. ID: " + positionId);
        logger.error("Position not found. ID: " + positionId);

        this.code = "POSITION_NOT_FOUND";
    }

    public String getCode() {
        return code;
    }

}

File: project/recruitment/frontend/recruitment-bff/src/positions/model/exceptions/position-not-found.exception.ts
import { BaseException } from '../../../shared/exceptions/base.exception';

export class PositionNotFoundException extends BaseException {
  constructor(positionId: number) {
    super(1000, `Position with id ${positionId} not found`);
    this.logger.error(`Position with id ${positionId} not found`);
  }
}